vendor/k8s.io/kubernetes/pkg/kubelet/container/helpers.go

```diff
@@ -302,7 +302,7 @@ func GetContainerSpec(pod *v1.Pod, containerName string) *v1.Container {
 
 // HasPrivilegedContainer returns true if any of the containers in the pod are privileged.
 func HasPrivilegedContainer(pod *v1.Pod) bool {
-	for _, c := range pod.Spec.Containers {
+	for _, c := range append(pod.Spec.Containers, pod.Spec.InitContainers...) {
```

Comment:
This change seems necesary but I don't *think* the container where we're seeing the rejection in the bug is an init container(we do have privileged init containers in our build pods but they only come into play in certain circumstances), so i'm uncertain this will actually fix the issue we're seeing.

https://github.com/openshift/origin/blob/master/pkg/build/controller/strategy/sti.go#L73-L177

Basically we use privileged init containers to pull images if you have an "image source input" in your build.

the build in bug https://bugzilla.redhat.com/show_bug.cgi?id=1550015 would not be doing that:
`oc new-app --template=nodejs-mongo-persistent`

so i think it is failing in the main container, as defined here:
https://github.com/openshift/origin/blob/master/pkg/build/controller/strategy/sti.go#L81-L101

Reply:
also since our main container is always privileged, this change wouldn't have any effect on whether this function returns true or false for our pods (it would have returned true before, and it will still return true)

Reply:
Yeah that's the one we see failing. Need to debug why the privileged isn't taking effect.